src/Hpp/Tokens.hs

```diff
@@ -95,16 +96,26 @@ tokWords s =
             '\\' :. cs ->
               case sbreak (boolJust . (== '\'')) cs of
                 Nothing -> [Important (pre' <> pos)]
-                Just (_,esc,pos') ->
-                  let esc' = if sall isOctDigit esc
-                             then Important (digitsFromBase 8 esc)
-                             else case esc of
-                                    'x' :. hs
-                                      | sall isHexDigit hs ->
-                                      Important (digitsFromBase 16 hs)
-                                    (escapeChar -> Just e) :. Nil -> Important e
-                                    _ -> Important ("'\\" <> snoc esc '\'')
-                  in  maybeImp pre ++ esc' : tokWords pos'
+                Just (_,esc,pos')
```

Comment:
This fixes the issue I mentioned where character literals are mutated into their numeric representations.